WordPress/gutenberg
# Block Supports: Add border color, style and width support

Fixes: #29616

## Description
This PR extends the border related block support to include color, style and width options.

These border options are disabled by default.

## How has this been tested?

Manually with a couple of small updates to existing unit tests.

#### Testing Instructions:

1. Enable custom border color, radius, style and width in your theme.json file.
2. Create a new post and add a group block to it
3. Select the group block and confirm the "Border settings" panel displays in the sidebar
4. Expand the "Border settings" panel and try editing each of the border properties. 
5. Confirm the block updates as expected in the editor, save and then confirm correct classes and styles on the frontend.
6. Open the Site Editor then its global styles sidebar. Switch to the "By block type" tab and expand the Group block panel.
7. Confirm that the border support features you have enabled are present. 
8. Test changing the global styles for the group block's borders.

#### Unit Tests:
- `npm run test-unit:watch packages/block-editor/src/hooks/test/style.js`
- `wp-env run phpunit "phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist /var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php"`

## Screenshots 
<img src="https://user-images.githubusercontent.com/60436221/112101007-b7164000-8bf1-11eb-9bb0-168af504cf15.gif" width="400" />

## Types of changes
New feature

## Next steps

- [ ] Make any required UI refinements
- [ ] Confirm approach to border color support
- [ ] Surface the border color option under the colors panel as well if desired (https://github.com/WordPress/gutenberg/issues/27331#issuecomment-793799694)

## Checklist:
